include/itkAttenuationImageFilter.h

```diff
@@ -0,0 +1,204 @@
+#include "itkMacro.h"
+
+#include "itkNumericTraits.h"
+
+namespace itk
+{
+/** \class AttenuationImageFilter
+ * \brief Computes the estimated attentuation in dB/(MHz*cm)
+ *
+ * Image sink receives an image and outputs estimated attenuation.
+ *
+ * \sa VariableLengthVector
+ *
+ * \ingroup Operators
+ * \ingroup ITKImageStatistics
+ * \ingroup Ultrasound
+ */
+template <typename TInputImage,
+          typename TOutputImage = itk::Image<double, TInputImage::ImageDimension>,
+          typename TMaskImage = itk::Image<unsigned char, TInputImage::ImageDimension>>
+class ITK_TEMPLATE_EXPORT AttenuationImageFilter : public ImageToImageFilter<TInputImage, TOutputImage>
+{
+public:
+  ITK_DISALLOW_COPY_AND_MOVE(AttenuationImageFilter);
+
+  /** Standard class type aliases. */
+  using Self = AttenuationImageFilter;
+  using Superclass = ImageToImageFilter<TInputImage, TOutputImage>;
+  using Pointer = SmartPointer<Self>;
+  using ConstPointer = SmartPointer<const Self>;
+
+  /** Method for creation through the object factory. */
+  itkNewMacro(Self);
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro(AttenuationImageFilter, ImageToImageFilter);
+
+  /** Image type alias support */
+  static constexpr unsigned int ImageDimension = TInputImage::ImageDimension;
+
+  using InputImageType = TInputImage;
+  using InputImagePointer = typename TInputImage::Pointer;
+
+  using OutputImageType = TOutputImage;
+  using OutputImagePointer = typename TOutputImage::Pointer;
+
+  using MaskImageType = TMaskImage;
+  using MaskImagePointer = typename TMaskImage::Pointer;
+  using MaskPixelType = typename TMaskImage::PixelType;
+
+  using InputRegionType = typename TInputImage::RegionType;
+  using InputSizeType = typename TInputImage::SizeType;
+  using InputIndexType = typename TInputImage::IndexType;
+  using InputPixelType = typename TInputImage::PixelType;
+
+  using OutputRegionType = typename TOutputImage::RegionType;
+  using OutputPixelType = typename TOutputImage::PixelType;
+
+  itkSetInputMacro(MaskImage, TMaskImage);
+  itkGetInputMacro(MaskImage, TMaskImage);
+
+  itkGetConstMacro(LabelValue, MaskPixelType);
+  itkSetMacro(LabelValue, MaskPixelType);
+
+  itkGetConstMacro(Direction, unsigned int);
+  itkSetMacro(Direction, unsigned int);
+
+  itkGetConstMacro(SamplingFrequencyMHz, float);
+  itkSetMacro(SamplingFrequencyMHz, float);
+
+  itkGetConstMacro(FrequencyBandStartMHz, float);
+  itkSetMacro(FrequencyBandStartMHz, float);
+
+  itkGetConstMacro(FrequencyBandEndMHz, float);
+  itkSetMacro(FrequencyBandEndMHz, float);
+
+  itkGetConstMacro(ConsiderNegativeAttenuations, bool);
+  itkSetMacro(ConsiderNegativeAttenuations, bool);
+
+  itkGetConstMacro(PadPixelUpperBounds, unsigned int);
+  itkSetMacro(PadPixelUpperBounds, unsigned int);
+  itkGetConstMacro(PadPixelLowerBounds, unsigned int);
+  itkSetMacro(PadPixelLowerBounds, unsigned int);
+
+  itkGetConstMacro(PadSpatialUpperBoundsMM, float);
```

Comment:
It looks like you misunderstood this "fixed attenuation depth". There shouldn't be two ivars, just one `m_FixedEstimationDepth`. That means that pixel pairs should have constant distance between them, e.g. 3 mm.

Reply:
Ah okay, I did not understand that, thanks. Can you please clarify, is pairwise estimation depth fixed in addition to pixel/spatial padding at either end of an inclusion, or did I misunderstand the need for padding?

Reply:
Fixed depth is independent from padding, and it is an additional feature.